frontend/src/components/TE/TaskMixin.ts

```diff
@@ -1,8 +1,37 @@
 import { Component, Vue } from "vue-property-decorator";
 import _ from "lodash";
+import firebase from "firebase/app";
+import "firebase/functions";
 
 @Component
 export default class TaskMixin extends Vue {
+  cancelData() {
+    return {
+      status: "Spare",
+      timestampGiven: "",
+      assignee: {
+        emailAddress: "",
+        name: ""
+      }
+    };
+  }
+
+  async cancelAllotment(
+    item: any,
+    itemPath: string,
+    paths: any,
+    fieldUpdates: any
+  ) {
+    try {
+      await firebase.functions().httpsCallable("TE-cancelAllotment")({
+        taskId: item[".key"]
+      });
+      this.updateFields(item, fieldUpdates);
```

Comment:
We don't need to update the database from the frontend. It is supposed that all this is done by the function. So you need to call the function only.

Reply:
That function only updates the local state on a successful https function call.

Reply:
Ok then